### ISSUE TITLE
Comments in manifest.json not recognized

### ISSUE THREAD

**Issue by rugk** [hbenl/vscode-firefox-debug#91]
This `launch.json` is being rejected:

```
{
    "version": "0.2.0",
    "configurations": [
      {
        "name": "Launch addon",
        "type": "firefox",
        "request": "attach",
        "addonType": "webExtension",
        "addonPath": "${workspaceFolder}/src"
      }
    ]
}
```

![laucnherror](https://user-images.githubusercontent.com/11966684/46677079-2b0b2c00-cbe2-11e8-9890-9a6515574988.png)

It seems to care about the slash in `/src`, but well… this is where this dir is and I have to specify it.

Linux, Fedora 28
Code – OSS 1.26

**Comment by hbenl** [comment]
This is not a problem in `launch.json` but rather in `manifest.json`: That file contains a comment, which is not allowed in JSON:
![screenshot_2018-10-09_17-48-08](https://user-images.githubusercontent.com/16060807/46681668-074ce380-cbec-11e8-9b4e-204b035201fb.png)

**Comment by rugk** [comment]
Ouuh… where can I report this?
So is this a VS code issue or one of your add-on?

Because as I had to report to, seemingly thousand (:wink:) projects, it is actually allowed in the `manifest.json`:
* https://discourse.mozilla.org/t/manifest-json-should-mention-it-allows-comments/27529
* https://github.com/WeblateOrg/weblate/issues/2043
* https://github.com/Lusito/web-ext-translator/issues/10
* https://github.com/mozilla/web-ext/issues/1352
* …

**Comment by hbenl** [comment]
In that case it's an issue with my extension.

**Comment by rugk** [comment]
BTW what made me see the issue in the wrong file is obviously that the dialog links to the `launch.json`, which is the wrong file, which does not even have this issue. If you have any influence on that, you may change that, too. :smile:

**Comment by hbenl** [comment]
Fixed in version 1.7.1

> the dialog links to the `launch.json`

I couldn't make that button disappear, but I changed the error message, it now starts with "Couldn't parse manifest.json", which should be clear enough.
Also, VS Code will still complain about the comments in `manifest.json` (but you can ignore that, it won't do any harm) because it doesn't know that this is no regular JSON file. I think it's possible to tell Code that this is a `jsonc` (JSON with comments) file, but I didn't investigate any further (I would accept a PR, of course :wink:).

**Comment by rugk** [comment]
> (but you can ignore that, it won't do any harm) 

Yeah, or you njust change the file type to "JSONm", which is VSCode's term for "JSOn with comments". Don't know if this can be done permanently, though. :smile:
